src/lightning/data/streaming/dataset.py

```diff
@@ -173,3 +155,11 @@ def __next__(self) -> Any:
         self.index += 1
 
         return data
+
+
+def _try_create_cache_dir() -> Optional[str]:
+    if "LIGHTNING_CLUSTER_ID" not in os.environ or "LIGHTNING_CLOUD_PROJECT_ID" not in os.environ:
+        return None
```

Comment:
Refactored this logic because there is no need to get the actual values from the environment